week2/homework/src/counter.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const fs = require('fs').promises;
+const path = require('path');
+const express = require('express');
+
+const app = express();
+app.set('port', process.env.PORT || 3000);
+
+const PORT = app.get('port');
+const FILE = path.join(__dirname, process.env.FILE || 'counters.json');
+
+const loadCounters = async () => {
+  try {
+    const counters = await fs.readFile(FILE, 'utf8');
+    return JSON.parse(counters);
+  } catch (e) {
+    return [];
+  }
+};
+
+const saveCounters = async counters => {
+  try {
+    return await fs.writeFile(FILE, JSON.stringify(counters));
+  } catch (e) {
+    return [];
+  }
+};
+
+const listCounters = async () => {
+  try {
+    return await loadCounters();
+  } catch (e) {
+    throw new Error('Please, refresh the page!');
```

Comment:
you are throwing an error in your catch statement,  this may cause a server failure and it will make it stop in nodejs if for some reason you forgot to catch your errors.